Userland/Libraries/LibCore/System.cpp

```diff
@@ -178,6 +179,64 @@ ErrorOr<void> profiling_free_buffer(pid_t pid)
     int rc = ::profiling_free_buffer(pid);
     HANDLE_SYSCALL_RETURN_VALUE("profiling_free_buffer", rc, {});
 }
+
+ErrorOr<void> exec(StringView filename, Span<StringView> arguments, SearchInPATH search_in_path, Optional<Span<StringView>> environment)
+{
+    auto copy_strings = [](Span<StringView> const& source, Syscall::StringListArgument& dest) {
+        dest.strings = new Syscall::StringArgument[source.size()];
+        dest.length = source.size();
+        for (size_t i = 0; i < source.size(); ++i) {
+            dest.strings[i] = { source[i].characters_without_null_termination(), source[i].length() };
+        }
+    };
+
+    Syscall::SC_execve_params params;
+    copy_strings(arguments, params.arguments);
+
+    if (environment.has_value()) {
+        copy_strings(environment.value(), params.environment);
+    } else {
+        size_t env_count = 0;
+        for (size_t i = 0; environ[i]; ++i)
+            ++env_count;
+        params.environment.strings = new Syscall::StringArgument[env_count];
+        params.environment.length = env_count;
+        for (size_t i = 0; i < env_count; ++i) {
+            params.environment.strings[i] = { environ[i], strlen(environ[i]) };
+        }
+    }
+
+    auto run_exec = [](Syscall::SC_execve_params& params) -> ErrorOr<void> {
+        int rc = syscall(Syscall::SC_execve, &params);
```

Comment:
Do we get anything from re-implementing the conversion to syscall again in `LibCore` instead of just forwarding the arguments to the `LibC` implementation of `execve(..)` or `execvpe(..)` depending on the arguments? Looking at the code in `LibC` vs this I don't see any obvious benifit from the duplication, but perhaps I'm missing something. 

Forwarding to `LibC` would also allow us still use this from `Lagom` on other platforms, which is a pattern the rest of the these `System::` API's currently try to support.

Reply:
I've rewritten it to forward to the LibC functions. The downside is not being able to pass StringViews, since we need the strings to be null-terminated. That means some users have to copy their string data if it wasn't Strings already, but that shouldn't be a performance issue anyway.